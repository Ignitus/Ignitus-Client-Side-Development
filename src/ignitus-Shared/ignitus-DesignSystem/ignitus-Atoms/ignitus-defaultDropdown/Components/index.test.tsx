import React from 'react';
import { mount, render } from 'enzyme';
import emotionSerialize, { matchers } from 'jest-emotion';

import { DefaultDropdown } from '.';

expect.addSnapshotSerializer(emotionSerialize);
expect.extend(matchers);

const props = {
  options: ['option1', 'option2', 'option3', 'option4'],
  label: 'options',
  display: 'block',
};

it('should rendered', () => {
  const tree = render(<DefaultDropdown {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<DefaultDropdown />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<DefaultDropdown {...props} />);
  });

  it('should render label, select & option tags', () => {
    expect(wrapper.find('label').length).toBe(1);
    expect(wrapper.find('select').length).toBe(1);
    expect(wrapper.find('option').length).toBe(props.options.length);
  });

  it('should render select with passed display option', () => {
    expect(wrapper.find('select')).toHaveStyleRule('display', props.display);

    wrapper = mount(<DefaultDropdown {...{ ...props, display: 'inline' }} />);
    expect(wrapper.find('select')).toHaveStyleRule('display', 'inline');
  });
});
