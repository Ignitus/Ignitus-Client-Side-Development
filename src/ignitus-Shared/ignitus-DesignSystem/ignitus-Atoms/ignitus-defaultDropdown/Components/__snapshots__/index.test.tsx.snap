// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered 1`] = `
Array [
  <label
    for="options"
  >
    options *
  </label>,
  .emotion-0 {
  display: block;
  margin-left: 0;
  background-color: #ffffff;
  border: 2px solid #000066;
  border-radius: 12px;
  font-size: 14px;
  height: 2rem;
  width: 18rem;
  outline: none;
}

<select
    class="emotion-0"
    display="block"
    id="options"
  >
    <option
      value="option1"
    >
      option1
    </option>
    <option
      value="option2"
    >
      option2
    </option>
    <option
      value="option3"
    >
      option3
    </option>
    <option
      value="option4"
    >
      option4
    </option>
  </select>,
]
`;
